source/particle/property/interface.cc

```diff
@@ -286,6 +286,56 @@ namespace aspect
 
 
 
+      template <int dim>
+      void
+      IntegratorProperties<dim>::initialize_one_particle_property(const Point<dim> &/*position*/,
+                                                                  std::vector<double> &data) const
+      {
+        data.resize(data.size() + n_integrator_properties);
+      }
+
+
+
+      template <int dim>
+      std::vector<std::pair<std::string, unsigned int> >
+      IntegratorProperties<dim>::get_property_information() const
+      {
+        const std::vector<std::pair<std::string,unsigned int> > property_information (1,
+                                                                                      std::make_pair("integrator properties",n_integrator_properties));
+        return property_information;
+      }
+
+
+
+      template <int dim>
+      void
+      IntegratorProperties<dim>::parse_parameters (ParameterHandler &prm)
+      {
+        std::string name;
+        prm.enter_subsection ("Postprocess");
+        {
+          prm.enter_subsection ("Particles");
+          {
+            name = prm.get ("Integration scheme");
+
+            if (name == "rk2")
+              n_integrator_properties = dim;
+            else if (name == "rk4")
+              n_integrator_properties = 4*dim;
+            else if (name == "euler")
+              n_integrator_properties = 0;
```

Comment:
Do we need to know this at this point already, or could we let the derived classes figure this out themselves and pass to the constructor? It's a quantity that is known at object-creation time. We just need to create the integrators either here or in the function that calls this place. Point being that we shouldn't have to figure this out here, but the integrators should say themselves how much internal space they need.

Reply:
I would agree, unfortunately they depend on each other. Integrators are not yet created at this point, because when they are created they need to know where the propertymanager class stores their internal properties. So one of them has to go first, and it is easier for the property manager to figure out how many properties are stored by the integrator, than it is for the integrator to figure out at which position the integrator properties are going to be stored. This is also why I cant make this property `const` at the moment.

Reply:
That's a bummer. This place knows the integrator classes, though. Can we make the number of intermediate properties a `static const` member variable of these classes, and use symbolic names here? In other words, let the integrator classes provide this information so one isn't left wondering where these numbers come from.

Reply:
yup that works. I changed it.